### PR TITLE
Check in changes to Support the Project and allow for none Standerd …

### PR DESCRIPTION
**Pull request by majianlord**
This is to support the Patch and other none Default verbs.  They must be sent as attributes since the AJP13 process does not have a byte configuration for that method.  Instead you sent them as Method byte 255 and then add the attribute with the value of the verb.  I have confirmed this working in my systems.   This is a fix for Issue Http Verb Issue #40

### CODE REVIEW

**Comment by majianlord** [comment]
Awsome

**Comment by Bilal-S** [comment]
@majianlord thanks very much  for the PR submission.
I will look through on the weekend.

File: BonCodeAJP13/ServerPackets/BonCodeAJP13ForwardRequest.cs [hunk 1]
```diff
@@ -566,10 +566,19 @@ private void WritePacket(NameValueCollection httpHeaders, String pathInfo, int s
...
+
             //WRITE NAMED ATTRIBUTES
 
 
+            //Check if the METHOD needs to be sent in the STORED METHOD Attribute IF SO we Set that Attribute and Value
+            if (method == BonCodeAJP13HTTPMethods.BONCODEAJP13_SC_M_JKSTORED)
+            {
+                string NonStandardVerb = GetKeyValue(httpHeaders, "REQUEST_METHOD");
+                pos = SetByte(aUserData, BonCodeAJP13HTTPAttributes.BONCODEAJP13_STORED_METHOD, pos); //attribute marker
+                pos = SetString(aUserData, NonStandardVerb, pos);  //method: e.g. we have clicked on URL
```
Comment (Bilal-S) [inline BonCodeAJP13/ServerPackets/BonCodeAJP13ForwardRequest.cs:578]: line 578 - the example in comments should be probably something else, such `// override method: such as PATCH`

**Review by Bilal-S — CHANGES_REQUESTED** [review]
@majianlord Thanks for the changes  looks good only minor adjustments and one question.  If you could make the few adjustments I will merge this and also have a few additional changes that need to be published.

File: BonCodeAJP13/BonCodeAJP13Enums.cs [hunk 2]
```diff
@@ -141,6 +141,7 @@ public struct BonCodeAJP13HTTPMethods
         public const byte BONCODEAJP13_MERGE = 0x19;            // MERGE 25 
         public const byte BONCODEAJP13_BASELINE_CONTROL = 0x1A; // BASELINE_CONTROL 26 
         public const byte BONCODEAJP13_MKACTIVITY = 0x1B;       // MKACTIVITY 27 
+        public const byte BONCODEAJP13_SC_M_JKSTORED = 0xFF;       // Any other VERB 255 to Stroe in Attributes
```
[the remark this answers is not in the document]
Comment (majianlord) [reply BonCodeAJP13/BonCodeAJP13Enums.cs:144]: Even though it's not well-documented on the [official page](https://tomcat.apache.org/connectors-doc/ajp/ajpv13a.html), this functionality has actually been around since at least 2000.

Tracked a bug ([56884](https://bz.apache.org/bugzilla/show_bug.cgi?id=56884)) and peeked at the code documentation for both [Tomcat 10](https://tomcat.apache.org/tomcat-10.0-doc/api/org/apache/coyote/ajp/Constants.html#SC_M_JK_STORED) and [Tomcat 7](https://tomcat.apache.org/tomcat-7.0-doc/api/org/apache/coyote/ajp/Constants.html#SC_M_JK_STORED), as well as the [Tomcat code](https://github.com/apache/tomcat) itself.

In a nutshell: By default, the method looks in the passed-over method slot and saves a few bytes for pre-defined methods. There’s a fallback that tells the protocol to look in the attributes for the method instead. Validated and working from Tomcat 7 to 10.1.

I have been managing about 100 Tomcat apps behind IIS and a lot of them use REST calls with patch, which led me down this rabbit hole.  Looks like the original protocol designers had a solution, just didn’t document it too well.  I am happy to add to the comment but pretty sure it would not be accurate.

File: BonCodeAJP13/ServerPackets/BonCodeAJP13ForwardRequest.cs
[the remark this answers is not in the document]
Comment (majianlord) [reply BonCodeAJP13/ServerPackets/BonCodeAJP13ForwardRequest.cs:1]: 338,  Yeah Uh i need to think on that the best way to handle that for logging.
403,  Yes i have been able to verify,  Running on over 100 Different IIS to Tomcat Instances with versions 7 to 10.1  The Method does Exisit just not documented in the easy to find documentation.    I dont know about the priority which takes precidents i will do some more testing and comment back here

File: BonCodeAJP13/ServerPackets/BonCodeAJP13ForwardRequest.cs (hunk 1 above)
Comment (majianlord) [reply BonCodeAJP13/ServerPackets/BonCodeAJP13ForwardRequest.cs:578]: How does this sound. 
//method to handle the passing of an arbitrary HTTP methods not handled by AJP 1.3 default binary methods

File: BonCodeAJP13/BonCodeAJP13Enums.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (majianlord) [reply BonCodeAJP13/BonCodeAJP13Enums.cs:144]: public const byte BONCODEAJP13_SC_M_JKSTORED = 0xFF;    // Any other VERB 255
//Binary Flag to signal the use of ?stored_method in Attributes
//https://tomcat.apache.org/tomcat-10.0-doc/api/org/apache/coyote/ajp/Constants.html#SC_M_JK_STORED

How about that for an Updated Comment.

[the remark this answers is not in the document]
Comment (Bilal-S) [reply BonCodeAJP13/BonCodeAJP13Enums.cs:144]: Thanks for the research. The last comment is good.

File: BonCodeAJP13/ServerPackets/BonCodeAJP13ForwardRequest.cs
[the remark this answers is not in the document]
Comment (Bilal-S) [reply BonCodeAJP13/ServerPackets/BonCodeAJP13ForwardRequest.cs:1]: ok. Thanks for checking on 403. 
Feel free to propose a change for 338 and let me know. E.g. there is a piece of commented out code that would be ok.

File: BonCodeAJP13/ServerPackets/BonCodeAJP13ForwardRequest.cs (hunk 1 above)
Comment (Bilal-S) [reply BonCodeAJP13/ServerPackets/BonCodeAJP13ForwardRequest.cs:578]: Yes that will work.

**Comment by Bilal-S** [comment]
Will resolve comments and changes in separate push
